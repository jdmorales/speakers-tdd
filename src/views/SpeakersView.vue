<template>
  <div>
    <SearchForm/>
    <SpeakersList :speakers="speakers"/>
  </div>
</template>

<script>
import SearchForm from '@/components/SearchForm'
import SpeakersList from '@/components/SpeakersList'
import { mapGetters } from 'vuex'

export default {
  name: 'SpeakersView',
  components: {
    SearchForm,
    SpeakersList
  },
  computed: {
    ...mapGetters({speakers: 'speakers'})
  }
}
</script>