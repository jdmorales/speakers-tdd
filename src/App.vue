<template>
  <div id="app">
    <SpeakersView />
  </div>
</template>

<script>
import SpeakersView from '@/views/SpeakersView'
export default {
  name: "app",
  components: {
    SpeakersView
  }
};
</script>
