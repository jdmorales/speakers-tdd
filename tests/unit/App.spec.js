import { shallowMount } from '@vue/test-utils'
import App from '@/App'
import SpeakersView from '@/views/SpeakersView'

describe('App', () => {

  const build = () => {
    const wrapper = shallowMount(App)

    return {
      wrapper,
      SpeakersView: () => wrapper.find(SpeakersView)
    }
  }

  it('renders the component', () => {
    // arrange
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the right', () => {
    // arrange
    const { SpeakersView } = build()
    // assert
    expect(SpeakersView().exists()).toBe(true)
  })
})