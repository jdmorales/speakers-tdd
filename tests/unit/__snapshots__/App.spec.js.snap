// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the component 1`] = `
<div id="app">
  <speakersview-stub></speakersview-stub>
</div>
`;
